File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n = nums.size();

        if(n==0){
            return 0;
        }
        sort(nums.begin(), nums.end());
        int count = 1;
        int maxcount = 1;
       
        for(int i=0; i<n-1; i++){
            if(nums[i]== nums[i+1]-1){
                count++;
                maxcount = max(maxcount, count);
            }

            else if(nums[i] == nums[i+1]){
                //do nothing
            }
            else{
                count = 1;
            }
        }
        return maxcount;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length == 0) return 0;
        Arrays.sort(nums);
        int longest = 1;
        int count = 0;
        int lastsmaller = Integer.MIN_VALUE;
        int n = nums.length;

        for(int i=0; i<n; i++){
            if(nums[i]-1 == lastsmaller){
                count = count+1;
                lastsmaller = nums[i];
            }
            else if(lastsmaller != nums[i]){
                count = 1;
                lastsmaller = nums[i];
            }
            longest = Math.max(longest, count);
        }
        return longest;
    }
}